### DISCUSSION TITLE
表格识别出现 list index out of range

### DISCUSSION THREAD

**Discussion by chenzhen1120** [PaddlePaddle/PaddleOCR#14249] [Q&A]
paddlepaddle：2.5.2
paddlehub:2.1.0
paddleocr版本试了几个都是这个错误
直接使用命名执行table/predict_table.py的话是可以正常推理的

**Comment by GreatV** [comment]
麻烦提供一个最小可复现demo

**Reply by chenzhen1120** [reply]
你好，麻烦看下

**Comment by chenzhen1120** [comment]
os: centos7
python:3.10

执行步骤
`pip install paddlepaddle==2.5.2 -i https://pypi.tuna.tsinghua.edu.cn/simple`

`pip install paddlehub==2.1.0 -i https://pypi.tuna.tsinghua.edu.cn/simple`

`git clone https://gitee.com/paddlepaddle/PaddleOCR.git`

`cd PaddleOCR`

`pip install -r requirements.txt -i https://pypi.tuna.tsinghua.edu.cn/simple`

`hub install deploy/hubserving/structure_table`

`nohup hub serving start -m structure_table -p 9161 &`


curl -X POST http://127.0.0.1:9161/predict/structure_table \
-H "Content-Type: application/json" \
-d '{"images": ['base64']}'

**Reply by GreatV** [reply]
直接使用python版本预测推理有问题吗

**Reply by chenzhen1120** [reply]
python table/predict_table.py --det_model_dir=../inference/ch_PP-OCRv4_det_infer --rec_model_dir=../inference/ch_PP-OCRv4_rec_infer --table_model_dir=../inference/ch_ppstructure_mobile_v2.0_SLANet_infer --rec_char_dict_path=../ppocr/utils/ppocr_keys_v1.txt --table_char_dict_path=../ppocr/utils/dict/table_structure_dict_ch.txt --image_dir=../images/table.png --output=../output/table  这样执行是没有问题的。

服务化之后出现
{
    "msg": "list index out of range",
    "results": "",
    "status": "101"
}

**Reply by GreatV** [reply]
paddlehub 好像不再维护了，可以考虑换其他方法进行服务化部署。

**Reply by GreatV** [reply]
https://paddlepaddle.github.io/PaddleX/latest/pipeline_usage/tutorials/ocr_pipelines/table_recognition.html#3 试试paddlex的服务化部署
